### ISSUE TITLE
Failed: Http2ServerSpec, connection failure

### ISSUE THREAD

**Issue by seglo** [akka/akka-http#3658]
https://jenkins.akka.io:8498/job/akka-http-nightly-release-10.1/AKKA_VERSION=master,Node=akka-http-nightly-node,SCALA_VERSION=2.13.1,jdk=AdoptOpenJDK 8/1089/testReport/akka.http.impl.engine.http2/Http2ServerSpec/The_Http_2_server_implementation_should_support_multiple_concurrent_substreams_should_close_substreams_when_connection_is_shutting_down/

```
sbt.ForkMain$ForkError: org.scalatest.exceptions.TestFailedException: "[Connection broke]" was not equal to "[The HTTP/2 connection was shut down while the request was still ongoing]"
	at org.scalatest.MatchersHelper$.indicateFailure(MatchersHelper.scala:343)
	at org.scalatest.Matchers$AnyShouldWrapper.shouldBe(Matchers.scala:6919)
	at akka.http.impl.engine.http2.Http2ServerSpec$$anon$35.<init>(Http2ServerSpec.scala:778)
	at akka.http.impl.engine.http2.Http2ServerSpec.$anonfun$new$48(Http2ServerSpec.scala:749)
	at akka.stream.testkit.scaladsl.StreamTestKit$.assertAllStagesStopped(StreamTestKit.scala:32)
	at akka.http.impl.engine.http2.Http2ServerSpec.$anonfun$new$47(Http2ServerSpec.scala:749)
	at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
	at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at org.scalatest.Transformer.apply(Transformer.scala:22)
	at org.scalatest.Transformer.apply(Transformer.scala:20)
	at org.scalatest.WordSpecLike$$anon$3.apply(WordSpecLike.scala:1075)
	at akka.http.impl.util.WithLogCapturing.$anonfun$withFixture$1(WithLogCapturing.scala:36)
	at akka.testkit.EventFilter.intercept(TestEventListener.scala:119)
	at akka.http.impl.util.WithLogCapturing.withFixture(WithLogCapturing.scala:34)
	at akka.http.impl.util.WithLogCapturing.withFixture$(WithLogCapturing.scala:21)
	at akka.http.impl.util.AkkaSpecWithMaterializer.withFixture(AkkaSpecWithMaterializer.scala:11)
	at org.scalatest.WordSpecLike.invokeWithFixture$1(WordSpecLike.scala:1073)
	at org.scalatest.WordSpecLike.$anonfun$runTest$1(WordSpecLike.scala:1085)
	at org.scalatest.SuperEngine.runTestImpl(Engine.scala:286)
	at org.scalatest.WordSpecLike.runTest(WordSpecLike.scala:1085)
	at org.scalatest.WordSpecLike.runTest$(WordSpecLike.scala:1067)
	at akka.testkit.AkkaSpec.runTest(AkkaSpec.scala:58)
	at org.scalatest.WordSpecLike.$anonfun$runTests$1(WordSpecLike.scala:1144)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:393)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:381)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:370)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:407)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:381)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:370)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:407)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:381)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:376)
	at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:458)
	at org.scalatest.WordSpecLike.runTests(WordSpecLike.scala:1144)
	at org.scalatest.WordSpecLike.runTests$(WordSpecLike.scala:1143)
	at akka.testkit.AkkaSpec.runTests(AkkaSpec.scala:58)
	at org.scalatest.Suite.run(Suite.scala:1124)
	at org.scalatest.Suite.run$(Suite.scala:1106)
	at akka.testkit.AkkaSpec.org$scalatest$WordSpecLike$$super$run(AkkaSpec.scala:58)
	at org.scalatest.WordSpecLike.$anonfun$run$1(WordSpecLike.scala:1189)
	at org.scalatest.SuperEngine.runImpl(Engine.scala:518)
	at org.scalatest.WordSpecLike.run(WordSpecLike.scala:1189)
	at org.scalatest.WordSpecLike.run$(WordSpecLike.scala:1187)
	at akka.testkit.AkkaSpec.org$scalatest$BeforeAndAfterAll$$super$run(AkkaSpec.scala:58)
	at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
	at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
	at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
	at akka.testkit.AkkaSpec.run(AkkaSpec.scala:58)
	at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:317)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:510)
	at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:304)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by johanandren** [comment]
Should be fixed by #3647

**Comment by johanandren** [comment]
Didn't happen again after that fix so I'll close this.
